index.js

```diff
@@ -1,6 +1,140 @@
-exports.build = function (options) {
-  // call flatpak build-init
-  // copy in files
-  // flatpak build-finish
-  console.log("NOT IMPLEMETED");
+'use strict';
+
+var _ = require('lodash');
+var path = require('path');
+var util = require('util');
+
+var promise = require('bluebird');
+var exec = promise.promisify(require('child_process').exec);
+var copy = promise.promisify(require('fs-extra').copy);
+var mkdirs = promise.promisify(require('fs-extra').mkdirs);
+var remove = promise.promisify(require('fs-extra').remove);
+var tmpdir = promise.promisify(require('tmp').dir);
+
+function addCommandLineOption (args, name, value) {
+    if (!value)
+        return;
+    args.push(util.format('--%s=%s', name, value));
+}
+
+function getOptionsWithDefaults (options) {
+    return _.defaults({}, options, {
+        branch: 'master',
+        sdk: 'org.freedesktop.Sdk',
+        runtime: 'org.freedesktop.Platform',
+        runtimeVersion: '1.4',
+        finishArgs: [],
+    });
+}
+
+function ensureDirectories (options) {
+    let dirs = [];
+    if (!options.buildDir) {
+        dirs.push(tmpdir().then(function (path) {
+            options.buildDir = path;
+        }));
+    } else {
+        dirs.push(remove(options.buildDir));
+    }
+    if (!options.repoDir)
+        dirs.push(tmpdir().then(function (path) {
+            options.repoDir = path;
+        }));
+    return promise.all(dirs).then(function () {
+        return options;
+    });
+}
+
+function flatpakBuildInit (options) {
+    var args = ['flatpak build-init'];
+    addCommandLineOption(args, 'arch', options.arch);
+    args.push(options.buildDir);
```

Comment:
I would make sure to shell-quote these args before joining them, in case they contain spaces. Kind of disappointing that `child_process.exec` takes a string instead of an args array...